### PR TITLE
Nonexistent event handlers now fail silently

### PR DESCRIPTION
**Pull request by paulfalgout**
Aligns with `Backbone` and `v3`.

Resolves https://github.com/marionettejs/backbone.marionette/issues/2888

### CODE REVIEW

**Comment by paulovieira** [comment]
Uhmm, this was actually a nice feature of marionette, i'm sad to see it go. You could save some precious debugging time if marionette tells you right away that the method is missing (a common mistake when the name of method is long).

I don't see much sense in imitating backbone just for the sake of it, especially the questionable parts.

What about add a "DEBUG" flag to the `Marionette` object and warn the user about this kind of problems when it is true? This would align with Radio:

https://github.com/marionettejs/backbone.radio/blob/master/src/backbone.radio.js#l21

**Comment by paulfalgout** [comment]
For almost any other hash, I think I'd agree.  I'm not sure the use-case for purposefully attempting to bind an event to a nonexistent function and having it `noop`.  Though I do wonder why it is not more consistent.

However I do think it would be fairly awkward in this circumstance considering it is essentially the exact same hash providing the same functionality for the view.

```js
var MyBehavior = Behavior.extend({
  events: function() {
    'click': 'missingFoo' // throw error?
  }
});

var MyBehavior2 = Behavior.extend({
  events: function() {
    return {
      'click': this.view.missingFoo // throw error?
    };
  }
});

var MyView = ItemView.extend({
  behaviors: {
    error: {
      behaviorClass: MyBehavior 
    },
    error2: {
      behaviorClass: MyBehavior2 
    }
  }
  events: {
    'click': 'missingFoo' // no error
  }
});

```

Here's backbone's rationale https://github.com/jashkenas/backbone/pull/2191
*TL;DR*
`$.on('test` 'missingFoo');` doesn't cause an error but `$.trigger('test');` will
The same is with `Backbone.Events.on` and `Backbone.View#events`
The error is thrown when triggered not when defined.

But I'm open to alternatives, and like the DEBUG idea.  I think in this case, this is fixing a bug in `2.4.4` but it'd be great to consider a new debugging feature for `v3`

**Comment by samccone** [comment]
+1 to warn on this one via a console && console.warn ? console.warn() : console.log()

**Comment by paulfalgout** [comment]
related: https://github.com/marionettejs/backbone.marionette/issues/1910

**Comment by paulovieira** [comment]
> Here's backbone's rationale jashkenas/backbone#2191
> 
> TL;DR
> $.on('test' 'missingFoo'); doesn't cause an error but $.trigger('test'); will
> The same is with Backbone.Events.on and Backbone.View#events
> The error is thrown when triggered not when defined.

Ah, interesting. It makes some sense from that point of view.

On the other hand, a more pragmatic approach would have value as well, that is, save developer's time by warning the user right away that something is missing.

Another option to consider if v3 were using webpack instead of gulp would be something like this:

**Comment by megawac** [comment]
Does this make sense in a patch release? I'd rather leave it. Also this needs unit tests, no?

**Comment by paulfalgout** [comment]
I think this PR fixes a bug.  Anything else is potentially a feature for later.  I can certainly add a test for this.

**Comment by megawac** [comment]
I'm = for this inclusion
